plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/SimpleUserNameProvider.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hive.metastore.thrift;
+
+import io.trino.spi.security.ConnectorIdentity;
+
+public class SimpleUserNameProvider
```

Comment:
This can be a singleton defined in the interface itself:
```
public static final SIMPLE_USER_NAME_PROVIDER = identity -> identity.getUser();
```
and you can bound it to instance in guice modules